File: go/game.go
package main

import (
	"fmt"
	"math/rand"
)

func NewGame() {
	target := rand.Intn(101)

	for {
		guess := ReadInt()

		if guess == target {
			fmt.Println("Congrats. You won")
			_, _ = fmt.Scan()
			return
		} else if guess > target {
			fmt.Println("Less")
			continue
		} else {
			fmt.Println("More")
			continue
		}
	}
}

File: rust_app/src/game.rs
use rand::Rng;
use crate::user_input;

pub fn new_game(min_val: u32, max_val: u32){
    let target: u32 = rand::thread_rng().gen_range(min_val..=max_val);

    loop {
        let guess: u32 = user_input::read_uint(&format!("Enter number from {min_val} and {max_val}"));

        if guess == target {
            println!("Congrats. You won");
            return
        } else if guess > target {
            println!("Less")
        } else {
            println!("More")
        }
    }
}